pkg/odo/cli/registry/registry.go

```diff
@@ -0,0 +1,226 @@
+			Header: text.FormatUpper,
+			Row:    text.FormatDefault,
+		},
+		Options: table.Options{
+			DrawBorder:      false,
+			SeparateColumns: false,
+			SeparateFooter:  false,
+			SeparateHeader:  false,
+			SeparateRows:    false,
+		},
+	})
+	t.SetOutputMirror(log.GetStdout())
+
+	t.AppendHeader(table.Row{"NAME", "REGISTRY", "DESCRIPTION"})
+	t.SortBy([]table.SortBy{
+		{Name: "NAME", Mode: table.Asc},
+		{Name: "REGISTRY", Mode: table.Asc},
+	})
+
+	devfiles := []registry.DevfileStack{}
+	// Filter through all the devfile components per the filters / parameters passed in.
+	for _, devfileComponent := range DevfileList {
+
+		// If the user has specified a filter with variable o.filterFlag, then only show the components
+		// containing that specific string.
+		if o.filterFlag != "" {
+			if !strings.Contains(devfileComponent.Name, o.filterFlag) && !strings.Contains(devfileComponent.Description, o.filterFlag) {
+				continue
+			}
+		}
+
+		// If the user passed in --devfile-registry <REGISTRY-NAME>, then only show the components from that Devfile stack
+		if o.registryFlag != "" {
+			if !strings.Contains(devfileComponent.Registry.Name, o.registryFlag) {
+				continue
+			}
+		}
+
+		// If the user passed in --devfile <NAME> only show that specific component matching that name
+		if o.devfileFlag != "" {
+			if devfileComponent.Name != o.devfileFlag {
+				continue
+			}
+		}
+		devfiles = append(devfiles, devfileComponent)
+	}
+
+	for _, devfileComponent := range devfiles {
+		// Mark the name as yellow in the index so it's easier to see.
+		name := text.Colors{text.FgHiYellow}.Sprint(devfileComponent.Name)
+
+		if o.detailsFlag {
+
+			// Output the details of the component
+			fmt.Printf(`%s: %s
+%s: %s
+%s: %s
+%s: %s
+%s: %s
+%s: %s 
+%s: %s
+%s: %s
+%s: %s
+%s:
+  - %s
+%s`,
+				log.Sbold("Name"), name,
+				log.Sbold("Display Name"), devfileComponent.DisplayName,
+				log.Sbold("Registry"), devfileComponent.Registry.Name,
+				log.Sbold("Registry URL"), devfileComponent.Registry.URL,
+				log.Sbold("Version"), devfileComponent.Version,
+				log.Sbold("Description"), devfileComponent.Description,
+				log.Sbold("Tags"), strings.Join(devfileComponent.Tags[:], ", "),
+				log.Sbold("Project Type"), devfileComponent.ProjectType,
+				log.Sbold("Language"), devfileComponent.Language,
+				log.Sbold("Starter Projects"), strings.Join(devfileComponent.StarterProjects, "\n  - "),
+				// TODO, showing dev / deploy / debug NOT yet implemented
+				// log.Sbold("Supported odo Features"), "Y", "Y", "Y",
+				"\n")
+		} else {
+			// Create a simplified row only showing the name, registry and description.
+			t.AppendRow(table.Row{name, devfileComponent.Registry.Name, util.TruncateString(devfileComponent.Description, 40, "...")})
+		}
+
+	}
+
+	// Exit with an error if there are no components to show, so we don't render the table / continue
+	if len(devfiles) == 0 {
+		log.Error("There are no Devfiles available to show")
+		return
+	}
```

Comment:
Can you move this before the for loop? I know for loop is not going to be bothered by 0 length, but it just makes sense to me to check the length and return early on if required.

Reply:
I put it after the for loop because the for loop ADDS to `devfiles` variable.